### PR TITLE
feat: restart/spawn worker panes without full orchestration/deck restart

### LINKED ISSUE #868: Auto-recover / re-spawn a crashed worker pane without manual TUI intervention
## Problem
When a role's pane (e.g. a Codex-backed `reviewer` role) crashes or dies, there is currently no way to detect and restart it from `worker-agent-deck`'s CLI surface. `delegate` only sends work to a pane that's already running — there's no `spawn`/`start`/`restart` subcommand. The only recovery path today is a human noticing from the TUI and manually restarting the pane, which:
- Stalls a multi-agent fix-and-recheck loop until a human notices and intervenes
- Is easy to miss if nobody is watching the TUI at that moment
- Gives the orchestrating agent no clear signal — a `delegate` call against a dead pane has no documented failure mode

## Context
Roles are defined in `.dot-agent-deck.toml`, e.g.:
```toml
[[orchestrations.roles]]
name = "reviewer"
command = "codex-devbox"
agent = "codex"
...
```
The underlying launch command works fine standalone (confirmed `codex-devbox` / `codex --version` both run correctly) — the crash is at the pane/session-management layer, not the underlying agent binary.

## Proposal
1. Add a subcommand (e.g. `worker-agent-deck pane restart <role>` or `pane spawn <role>`) that (re)starts a role's pane from its toml definition, callable from an orchestrating agent's own CLI flow as well as the TUI.
2. `delegate` should detect a dead target pane and report a clear, actionable error instead of silently no-op'ing or hanging — ideally naming the restart command to run.
3. Optional: a per-role or mode-wide `auto_restart` config flag so a crashed pane respawns automatically, similar to how sessions already restore prompt/seed state.

## Motivating incident
While running a Codex-backed `reviewer` role for a cross-agent SDD verification pass, the pane died mid-task. The orchestrating agent had no CLI-level way to detect this or restart it — recovery required the human to intervene directly in the TUI, stalling the review loop.

## Related
- #305 (Codex sandbox network access / delegate interaction)
- docs: troubleshooting#codex-as-a-role-or-worker-allow-sandbox-network-access

### PR DESCRIPTION
**Pull request by prageethw**
Closes #868.

## Problem

Today, if a worker/role pane in a running orchestration crashes, or a running orchestration needs one more role that wasn't part of its initial role set, the only recovery path is a human noticing from the TUI and manually restarting the pane or the whole tab. There's no CLI-level way for an orchestrating agent to detect this itself or fix it — a crashed worker just sits dead until someone looks, which stalls an unattended multi-agent fix-and-recheck loop and is easy to miss if nobody is watching.

This is exactly what #868 asks for.

## What this PR adds

Two new CLI subcommands, both scoped by `DOT_AGENT_DECK_PANE_ID` exactly like the existing `delegate`, so either a human or the orchestrating agent itself can call them from inside its own pane — no TUI keybind needed:

- **`dot-agent-deck pane restart <role>`** — restarts a worker role's pane within the calling orchestrator's own orchestration. Refuses a healthy pane unless `--force` is passed, so it can't be used to accidentally kill a working agent.
- **`dot-agent-deck pane spawn <role>`** — spawns a role that's declared in the orchestration's `.dot-agent-deck.toml` but wasn't yet running (e.g. a role added to the config mid-session), joining it into the already-open orchestration tab instead of requiring a full tab restart.

### Why this matters: a genuine self-healing loop

The daemon already delivers a notice (`compose_worker_exited_notice`) into the orchestrator's own pane when a delegated worker exits without `work-done`. Today nothing acts on that notice automatically. With this PR, the orchestrating agent can read that existing notice and call `pane restart <role>` itself — closing the loop without a human ever needing to look at the TUI.

### A deliberate difference from #868's own proposal

#868 also suggests an optional `auto_restart` config flag (daemon-side automatic restart policy). This PR doesn't add one, on purpose: since the daemon already tells the orchestrator a worker died, giving the *orchestrator* the restart primitive is enough — it can decide for itself whether and when to restart, without a separate daemon-side policy surface to configure and maintain. Happy to discuss if you'd still want the flag as a belt-and-suspenders option for orchestrators that don't act on the notice.

## Design notes for review

- Both new verbs ride the same unversioned hook-socket `DaemonMessage` channel `delegate`/`dispatch` already use (not `AttachRequest`), so no `PROTOCOL_VERSION` bump — same pattern the existing `GetSeed` verb already established.
- The new `AgentRecord.crashed` field is additive (`serde(default, skip_serializing_if)`), set only on a *natural* process exit (distinguished from a deliberate close/respawn via the same registry-membership check `pump_reader` already uses for that purpose).
- A crash-visibility fix in `TabManager` (`add_role_to_existing_orchestration`/`orchestration_tab_index_for`) makes a role spawned via `pane spawn` join an *already-open* orchestration tab rather than building a duplicate — including correct handling when two orchestration instances happen to share the same name/cwd (a per-tab `orchestration_id` token disambiguates them).
- Both new handlers (`handle_restart_role_with_state`/`handle_spawn_role_with_state`) are free functions taking `SharedState` directly rather than methods called through a pre-held read guard, so neither holds a lock across its (potentially multi-second) respawn/spawn — worth a look since the existing `Delegate` handler does hold its read guard across the whole call.

## Testing

Full new test coverage: `tests/pane_restart.rs` (6 tests), `tests/pane_spawn.rs` (9 tests), `tests/orchestration_tab_growth.rs` (2 TabManager-level tests), `tests/e2e_pane_spawn_live.rs` (2 PTY-attached e2e tests proving the tab-join behavior on a real spawned deck), plus unit coverage for the new `AgentRecord.crashed` field and wire-boundary validation. Ran locally against this branch (no CI on a personal branch yet):

- `cargo nextest run --workspace`: 2418/2418 passed.
- `cargo nextest run --workspace --features e2e`: 2607/2608 passed — the one failure (`orchestration_remit_007_compaction_reassertion_preserves_a_dispatched_task`) is a known, load-sensitive pre-existing flake (its own panic message cites #818); confirmed unrelated by running it in isolation (passes) and confirming this branch touches none of the files it exercises.

## Known, deliberately out-of-scope items

- No real-agent (LLM-driven) end-to-end test for either verb — coverage uses stand-in shell commands, consistent with most of the existing test suite for daemon/TUI logic.
- `pane spawn`'s CLI reply keeps the existing 5s `delegate`-shared timeout, while `pane restart`'s is sized up to ~14s to cover its own worst-case respawn grace period — a visible asymmetry, not a regression, happy to align them if you'd prefer.
- Process-exit crash detection only, not wedged-but-alive detection.
- `pane spawn` only targets a not-yet-live role already declared in the config — it doesn't create a true second pane under an already-running role name.

Let me know if you'd like anything reshaped before merge — happy to iterate.

### CODE REVIEW

**Review by vfarcic — CHANGES_REQUESTED** [review]
Verified this locally: clean merge, all four required checks green, `fmt`/`clippy`/`test-fast`/`linkage-check`/`windows-cross` all pass, and the new test suite is genuinely good — `pane/spawn/008`'s concurrent-spawn TOCTOU and `pane/spawn/010`'s same-`(name,cwd)` instance isolation in particular. Three things below need addressing before this goes in; two of them are not in the bot review.

## 1. Both verbs report success against an older daemon while doing nothing

Greptile flagged this on `src/main.rs`; I'm calling it out because I reproduced it rather than reasoned about it, and the result is worse than "degrades oddly".

Ran the rule-12 cross-version test — a real v0.39.3 daemon (`build_version 0.39.3-ga023377`) with both orchestration roles live under it, this branch's binary attached after declining the mismatch prompt:

```
$ dot-agent-deck pane restart coder    → exit 0, no stdout, no stderr
$ dot-agent-deck pane spawn reviewer   → exit 0, no stdout, no stderr
```

Daemon side at the same instant:

```
WARN Malformed event: {"message_type":"restart_role",...}
WARN Malformed event: {"message_type":"spawn_role",...}
```

`daemon status` confirmed nothing restarted and nothing spawned. It returns in ~4ms, so `RESTART_ROLE_REPLY_TIMEOUT` never comes into play — `serde` rejects the unknown variant at `src/daemon.rs:1825`, the daemon never replies, and `SocketReply::NoReply => ExitCode::SUCCESS` turns that into success.

The reason this matters more than an ordinary version-skew wrinkle: the documented primary consumer is the orchestrator agent calling this unattended ("The self-healing pattern"). Exit 0 with no output reads as "restarted", and it will then re-delegate into a pane that is still dead. `delegate`'s `NoReply => SUCCESS` is fine because delegate is fire-and-forget; these two are not.

Worth noting you already identified this exact failure shape in `RESTART_ROLE_REPLY_TIMEOUT`'s doc comment — "a timeout here silently maps to `ExitCode::SUCCESS` with no output ... while the outcome was genuinely unknown" — and widened the budget to cover the slow-daemon route. The unknown-verb route produces the identical symptom and isn't covered by that.

Suggest treating `NoReply` (and an unrecognised `kind`) as a distinct outcome for these two verbs: non-zero exit plus something like "the running daemon does not support `pane restart` — restart the daemon to pick up the new build."

## 2. The orchestrator is never told these commands exist

This one isn't in the bot review and I think it's the more consequential of the two.

```
files touching src/orchestrator_context.rs : 0
occurrences of "pane restart"/"pane spawn" in it : 0
```

`build_orchestrator_context()` is how an agent learns its CLI surface — it composes `.dot-agent-deck/orchestrator-context.md`, delivers it by pointer, carries the whole `## Delegation protocol` section with literal `{bin} delegate ...` invocations, and gets reasserted on compaction/clear via the #789 remit machinery. Nothing in this PR adds to it.

So the self-healing pattern the docs describe cannot happen by default. It works only if the user hand-writes the command into a `prompt_template` — which is exactly the workaround the new docs section proposes. The capability is reachable only by opting into hand-wiring.

Related, and the reason I'd fix these together: the CLI reference landed in the wrong channel. In `docs/orchestration.md` the established verbs appear as invocations *only* in Troubleshooting (`delegate`) and inside a `prompt_template` example (`work-done`); the main body is behavioural. The new `## Restarting and spawning worker panes` section documents both verbs with flag semantics in the main flow, which is a new pattern. Users should get the capability; agents should get the command.

What I'd suggest:

- Emit the two commands from `build_orchestrator_context`, kept terse — that section is already long and it costs context on every orchestrator turn.
- **Do not inject `--force`.** It exists to stop a healthy worker being killed mid-task, and an agent told about it will reach for it the moment the plain call is refused with "has not crashed". Keep `--force` in `--help` and Troubleshooting, for people.
- Shrink the docs section to the capability plus one `prompt_template` example, and move the flag/refusal detail to Troubleshooting. The example can then say the orchestrator is told automatically, rather than duplicating guidance that can drift from what is injected.
- Drop "Scope of this release" from the user docs — that's changelog material.

## 3. Rule 4 coverage

`pane/spawn/005` and `pane/drift/001` are real PTY-attached L2 tests and I confirmed they carry no skip guard and genuinely ran in CI. But:

- **`pane restart` has no PTY-attached test at all** — all six `pane/restart/*` entries are L1 handler calls, and the CATALOG records "Does not assert: the CLI/socket layer" for each. That's the verb named first in the PR title with no coverage of what a user sees.
- **No real-agent test for either verb.** Every new CATALOG entry is `Agent: none` with `cat`/`sleep 0.2` stand-ins, so nothing exercises the genuine spawn → agent → work path.

The one I'd insist on is a PTY-attached test for `restart` — it needs no credentials and no lane-2 machinery, and it's the half that's currently unobserved.

## Two things you do *not* need to act on

**`PROTOCOL_VERSION` is correct as it stands** — the bot's P2 is wrong and you can reply to it on that basis. `PROTOCOL_VERSION` versions the attach socket (`AttachRequest` / `KIND_*`), which this PR doesn't touch; `RestartRole`/`SpawnRole` ride the hook socket. Both new fields are additive `#[serde(default, skip_serializing_if = "Option::is_none")]` optionals, which the module header names as an explicit "Do NOT bump" case. I also verified the compatibility empirically in the cross-version run above: delegate still routes, and work-done and status hooks still arrive, against the v0.39.3 daemon.

**The two red e2e tests are not yours.** They're issue #818 (lane 1 red on every branch since #789), and they're one of each of its two causes: `form_007` is fixed by #872 (single-shot `find_in_grid` → polling `wait_for_in_grid`; it rewrites that exact line), `orchestration_remit_001` by #907 (load-scaled bounds). I confirmed it — form_007 passed 7/7 locally here, and remit_001 flips on the merge-base too (PASS/FAIL/PASS with none of your code). Note #907 also touches `src/ui.rs`, so whichever lands second may need a small rebase.

---

Filed #929 for the lazy-orchestration-start idea that came out of reading this — deliberately not a blocker on this PR, and your comment there would be welcome.

File: src/event.rs [hunk 1]
```diff
@@ -1187,6 +1301,15 @@ pub struct OrchestrationSurface {
     /// Optional user-facing tab title; `None` falls back to `name`.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub display_title: Option<String>,
+    /// Issue #868: the PRD #140 per-tab `Instance` orchestration
+    /// id, when the producer has one — additive, so an older daemon's surface
+    /// (or a `NameCwd`-identity orchestration, which has no such token)
+    /// carries `None`. `TabManager::orchestration_tab_index_for` requires this
+    /// to match a candidate tab's own stored id whenever BOTH sides carry one,
+    /// rather than falling back to the bare `(cwd, name)` tuple, which cannot
+    /// tell two same-named same-cwd orchestration instances apart.
```
[the remark this answers is not in the document]
Comment (vfarcic) [reply src/event.rs:1310]: Not applicable here — this is the documented *no-bump* case, and the rule is in the file the finding is about.

The field is added as:

```rust
#[serde(default, skip_serializing_if = "Option::is_none")]
pub orchestration_id: Option<String>,
```

`src/daemon_protocol.rs`'s versioning header states the policy explicitly:

> - **Do NOT bump:** additive optional fields tagged `#[serde(default, skip_serializing_if = "Option::is_none")]` — those are forward-compatible by design (older peer ignores the field, newer peer tolerates its absence).

When the value is `None` nothing is serialized at all, so an older daemon receives byte-identical frames; when it is present an older peer ignores it, and a newer peer tolerates its absence via `default`. No mis-parse is possible in either direction, which is the bump criterion the header names ("a change would cause an older or newer peer to mis-parse a frame").

This was also checked empirically, not just read: the cross-version manual test (CLAUDE.md rule 12) was run against the previous release with agents live under the older daemon, and delegate plus both hook paths still flowed.

`PROTOCOL_VERSION` is correct as-is. Resolving.

File: src/tab.rs [hunk 2]
```diff
@@ -1140,6 +1175,141 @@ impl TabManager {
...
+                max: crate::agent_pty::ORCHESTRATION_ROLE_INDEX_MAX,
+            });
+        }
+        config.roles.push(role_config);
+        role_pane_ids.push(pane_id);
+        // A freshly-spawned role is a live agent, not a dead slot — unlike
+        // `open_orchestration_tab_with_existing_role_panes`'s dead-slot
+        // handling, there's no `None` case here since M3's spawn only ever
+        // calls this on a real, just-spawned agent.
+        role_statuses.push(OrchestrationRoleStatus::Working);
```
[the remark this answers is not in the document]
Comment (vfarcic) [reply src/tab.rs:1251]: Accurate as a description of the behavior, but this is a known and explicitly documented limitation rather than an oversight.

The function's own doc comment in this PR says:

> A role inserted in the MIDDLE of the config, shifting existing role indices, is a known out-of-scope edge case for M4 — not handled here.

and `tests/CATALOG.md` records the same boundary under `pane/spawn/009`:

> **Does not assert:** a role inserted in the MIDDLE of the config, shifting existing role indices (a known out-of-scope edge case for M4).

The append and replace paths that M4 *does* claim are covered by `pane/spawn/009` and `pane/spawn/010`.

Resolving as declared out of scope for this PR. The mid-config insertion case is real and worth its own issue if we want it actually closed rather than documented.

File: src/main.rs [hunk 3]
```diff
@@ -1194,6 +1221,123 @@ fn main() -> ExitCode {
...
+                        eprintln!(
+                            "Error: could not reach the dot-agent-deck daemon socket, so the \
+                             restart of role {role} was NOT delivered."
+                        );
+                        return ExitCode::FAILURE;
+                    }
+                    // Same reasoning as `delegate`'s `NoReply`: a daemon that
+                    // doesn't know this verb yet (older build) must not read
+                    // as a proven failure.
+                    SocketReply::NoReply => return ExitCode::SUCCESS,
```
[the remark this answers is not in the document]
Comment (vfarcic) [reply src/main.rs:1264]: Confirmed — and not only by reading: this was reproduced directly against the previous release. A new-CLI `pane restart` / `pane spawn` issued to an already-running older daemon **exits 0 with no output while doing nothing**.

That is the feature's own headline unattended path failing silently: an orchestrator has no way to distinguish "worker restarted" from "the daemon never understood the request". Unlike `delegate`, neither verb was ever fire-and-forget, so a missing reply is not a benign case here.

This is carried as the blocking item in the review on this PR. Leaving the thread open until both verbs fail loudly on an unanswered or unrecognized request — the spawn path needs the same treatment at `src/main.rs:1317-1320`.

File: src/state.rs [hunk 4]
```diff
@@ -6415,6 +6402,510 @@ impl AppState {
...
+            role_index,
+            role_config,
+        }
+        // Read guard drops here — never held across the respawn below.
+    };
+
+    // The same dispatch lock `dispatch_one_owned` holds across its own
+    // respawn — see this function's doc comment.
+    let dispatch_mutex = registry.pane_dispatch_lock(&resolved.pane_id);
+    let _dispatch_guard = dispatch_mutex.lock().await;
```
[the remark this answers is not in the document]
Comment (vfarcic) [reply src/state.rs:6583]: Valid — confirmed structurally on review.

The non-force crash check does happen before this pane lock is acquired, so the interleaving described is reachable: a concurrent `clear = true` delegation can take the lock first and install a healthy replacement, after which this restart acquires the lock and respawns whatever now owns the pane without re-checking its generation or crash state. The healthy worker is killed without `--force`, and the command still reports a successful restart.

Leaving this thread open rather than resolving it, since it is a real unfixed defect. Clearing it wants either a fix in this PR — re-check generation/crash state *under* the lock rather than before it — or a tracking issue it can be resolved against.

**Comment by prageethw** [comment]
Thanks for the thorough review — all three items from the review body, plus the "Restart Can Kill Replacement" finding from the inline comments, are addressed now across two pushes:

## Review body items (all fixed in `02d9e7da`)

1. **Both verbs report success against an older daemon while doing nothing.** `pane restart`/`pane spawn` now exit non-zero on `SocketReply::NoReply` and on an unparseable reply, with a cause-agnostic message (doesn't flatly claim "old daemon" — a `DeadlineExpired` timeout can mean the operation already succeeded, so the message says either/or and tells the caller to check the pane before retrying rather than prescribing a daemon restart). `pane spawn` also got its own properly-sized reply timeout (was reusing `delegate`'s 5s budget, which is too tight now that a timeout is a hard failure).
2. **The orchestrator is never told these commands exist.** `build_orchestrator_context()` now emits a terse paragraph naming both commands — deliberately without `--force` (kept people-only, in `--help`/Troubleshooting), with a note that the daemon's own refusal message will still name `--force` if a restart is genuinely refused, so the docs don't overclaim total containment.
3. **No PTY-attached test for `pane restart`.** Added one, following `pane/spawn/005`'s technique (real TUI via PTY, real daemon, real CLI subprocess) — restarts a crashed worker and confirms a subsequent delegate still reaches it through the still-attached TUI.

Also restructured the "Restarting and spawning worker panes" docs section per your suggestions (shrunk to capability + one example, moved `--force`/refusal detail to Troubleshooting, dropped "Scope of this release" — though I kept the two genuinely-permanent limitations it contained, wedged-but-alive detection and the same-role-name restriction, as their own Troubleshooting entries rather than losing them entirely).

`PROTOCOL_VERSION`/the two red e2e tests: agreed, no action taken, per your own notes on both.

## Inline finding: "Restart Can Kill Replacement" (fixed in `92f9fc49`)

Confirmed and fixed as you described: `handle_restart_role_with_state` now re-checks crash/occupant state a second time, immediately after `pane_dispatch_lock` is acquired, before any respawn call — factored into a small shared helper so the check can't drift between the two call sites. If a concurrent `clear = true` delegation already installed a healthy replacement by the time this restart gets the lock, it now refuses with the same "has not crashed" error instead of killing the replacement. `--force` is unaffected by design (skips both checks, same as before). Added two tests: one pinning the race deterministically (holds the dispatch lock itself, drives the restart as a manually-polled future so the interleaving isn't timing-dependent, installs a replacement mid-flight, confirms the refusal), and one confirming `--force` still bypasses this new check specifically (not just the original one).

The other two inline findings (protocol version, tab.rs restore ordering) — agreed with your calls on both, no changes made.

One thing worth flagging: the branch shows `CONFLICTING`/`DIRTY` against current `main` now (expected, given how long this sat) — happy to rebase it if that's useful before you take another look, just say the word.

Prageeth Warnak

File: src/state.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (prageethw) [reply src/state.rs:7176]: Fixed in `92f9fc49` — `handle_restart_role_with_state` now re-checks crash/occupant state a second time right after `pane_dispatch_lock` is acquired, before the respawn. If a concurrent `clear = true` delegation already installed a healthy replacement by the time the lock is granted, this now refuses with the same "has not crashed" error instead of respawning it. `--force` is untouched — still bypasses both checks by design. Two new tests pin this: one holds the dispatch lock itself and drives the restart as a manually-polled future to prove the interleaving deterministically (not timing-based), and one confirms `--force` specifically bypasses the *new* post-lock check, not just the original one. Full detail in the PR-level comment above.

File: src/main.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (prageethw) [reply src/main.rs]: Fixed in `02d9e7da` — `pane restart`/`pane spawn` now exit `ExitCode::FAILURE` (not success) on `SocketReply::NoReply` and on an unparseable reply, with a cause-agnostic message (doesn't assert "old daemon" specifically, since a `DeadlineExpired` timeout can also mean the operation already succeeded — tells the caller to check the pane rather than prescribing a daemon restart). `pane spawn` also got its own 10s reply timeout (`SPAWN_ROLE_REPLY_TIMEOUT`) instead of reusing `delegate`'s 5s budget, since a timeout is now a hard failure rather than a benign fire-and-forget case. Two new CLI-level tests per verb drive the real binary against a stub daemon to cover both failure branches (`pane/restart/007`/`008`, `pane/spawn/012`/`013`). Full detail in the PR-level comment.

**Review by vfarcic — APPROVED** [review]
Approving — this supersedes my `CHANGES_REQUESTED` from 2026-09-06, which was stale rather than outstanding.

Both P1s from that review are genuinely fixed, verified against the code rather than taken on the strength of the resolved threads:

- **TOCTOU (`4a96dbe7`)** — the crash recheck is genuinely under `pane_dispatch_lock`, and it is load-bearing because `respawn_agent_for_pane_declared` removes the old registry entry and `spawn_agent` stamps a fresh one with `crashed: None` (the destructure at `src/agent_pty.rs:6807-6809` drops it deliberately), so the recheck reads the replacement's record and refuses. `dispatch_one_owned` takes the same lock as its first statement and holds it across its whole body, so nothing can be in flight once restart holds it. `pane/restart/010`/`011` pin both the refusal and the `--force` bypass with a manually-polled future rather than a scheduler race.
- **Exit codes (`40178393`)** — `NoReply` and an unparseable reply are both `ExitCode::FAILURE` for each verb, with cause-agnostic stderr that correctly does not assert "old daemon". The affirmative `kind` discriminator means an older daemon's arbitrary JSON cannot parse as a valid reply.

Rule 12 was answered correctly: both new fields are additive-optional `#[serde(default, skip_serializing_if = "Option::is_none")]`, which `src/daemon_protocol.rs`'s own versioning header names as the documented no-bump case. Greptile withdrew its P2 on this.

All 14 checks green on `4a96dbe7`, all seven local gates green (3118 tests). Two lane-2 real-agent tests were run by hand against this branch — `delegate_014_real_claude_worker_acts_on_clear_true_delegate` and `delegate_work_done_chain_claude`, both PASS — covering the shared respawn/EOF machinery this PR modifies.

Merging with three tracked follow-ups rather than blocking on them:

- **#1094** — neither new verb has a real-agent test, so CLAUDE.md rule 4's bar is knowingly deferred here, not overlooked. This is the one judgment call in the merge.
- **#1095** — `RESTART_ROLE_REPLY_TIMEOUT` is under-derived against the dispatch-lock wait.
- **#1096** — the mid-config-insert limitation, documented in-tree and now actually tracked.

Thanks for the thorough fix round — the test technique on `pane/restart/010`/`011` in particular is the right way to pin an interleaving.
